test/performance/scale_from_zero_test.go

```diff
@@ -71,24 +72,31 @@ func runScaleFromZero(idx int, t *testing.T, clients *test.Clients, ro *test.Res
 
 	start := time.Now()
 	t.Logf("%02d: waiting for endpoint to serve request", idx)
-	if _, err := pkgTest.WaitForEndpointStateWithTimeout(
+	resp, err := pkgTest.WaitForEndpointStateWithTimeout(
 		clients.KubeClient,
 		t.Logf,
 		domain,
 		pkgTest.MatchesAllOf(pkgTest.IsStatusOK, pkgTest.MatchesBody(helloWorldExpectedOutput)),
 		"HelloWorldServesText",
-		test.ServingFlags.ResolvableDomain, waitToServe); err != nil {
+		test.ServingFlags.ResolvableDomain, waitToServe)
+	if err != nil {
 		m := fmt.Sprintf("%02d: the endpoint for Route %q at domain %q didn't serve the expected text %q: %v", idx, ro.Route.Name, domain, helloWorldExpectedOutput, err)
 		t.Log(m)
 		return 0, errors.New(m)
 	}
-
+	dur := time.Since(start)
 	t.Logf("%02d: request completed", idx)
-	return time.Since(start), nil
+
+	// Sleep to get traces
+	time.Sleep(5 * time.Second)
```

Comment:
What do you think about moving this sleep to the `AddTrace` function? In my opinion, the sleep can easily be missed and `AddTrace` almost always need this extra delay.

Reply:
good point. Updated